Este host não é conhecido (elasticsearch)

Erro de conexão do Spring Boot com o ElasticSearch.

Mensagem de erro: Caused by: java.lang.RuntimeException: Este host não é conhecido (elasticsearch)

--- a/backend/abertoagora/Dockerfile
+++ b/backend/abertoagora/Dockerfile
@@ -2,5 +2,7 @@ FROM openjdk:21-slim
 WORKDIR /app
 RUN apt-get update && apt-get install -y curl
 COPY build/libs/abertoagora-0.0.1-SNAPSHOT.jar app.jar
-EXPOSE 8080
+COPY wait-for-it.sh /wait-for-it.sh
+RUN chmod +x /wait-for-it.sh
+EXPOSE 8081
 ENTRYPOINT ["java","-jar","/app/app.jar"]

--- a/backend/abertoagora/compose.yaml
+++ b/backend/abertoagora/compose.yaml
@@ -1,11 +1,8 @@
-version: '3.8'
-
 services:
   elasticsearch:
     image: 'docker.elastic.co/elasticsearch/elasticsearch:8.17.3'
-    container_name: elasticSearch
+    container_name: elasticsearch
     environment:
-      - 'ELASTIC_PASSWORD=AnSTXrcuYAL5kSc8Wsjq'
       - 'discovery.type=single-node'
       - 'xpack.security.enabled=false'
     ports:
@@ -14,9 +11,9 @@ services:
       - app-network
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:9200"]
-      interval: 10s
-      timeout: 5s
-      retries: 5
+      interval: 30s
+      timeout: 10s
+      retries: 3
 
   mysql:
     image: 'mysql:latest'
@@ -46,12 +43,10 @@ services:
       - 'spring.datasource.username=root'
       - 'spring.datasource.password=root'
       - 'spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver'
-      - 'spring.elasticsearch.uris=http://elasticsearch:9200'
-      - 'spring.elasticsearch.username=elastic'
-      - 'spring.elasticsearch.password=AnSTXrcuYAL5kSc8Wsjq'
+      - SPRING_ELASTICSEARCH_URIS=http://elasticsearch:9200
       - 'elasticsearch.enabled=true'
     ports:
-      - '8080:8080'
+      - '8081:8081'
     depends_on:
       elasticsearch:
         condition: service_healthy
@@ -59,7 +54,7 @@ services:
         condition: service_healthy
     networks:
       - app-network
-
+    command: /wait-for-it.sh elasticsearch:9200 -- java -jar /app.jar
 networks:
   app-network:
     driver: bridge

--- a/backend/abertoagora/src/main/java/br/com/abertoagora/controller/UsuarioController.java
+++ b/backend/abertoagora/src/main/java/br/com/abertoagora/controller/UsuarioController.java
@@ -1,0 +1,27 @@
+package br.com.abertoagora.controller;
+
+import br.com.abertoagora.dto.UsuarioDTO;
+import br.com.abertoagora.services.UsuarioServices;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/usuarios")
+public class UsuarioController {
+
+    private final UsuarioServices usuarioServices;
+
+    public UsuarioController(UsuarioServices usuarioServices) {
+        this.usuarioServices = usuarioServices;
+    }
+
+    @GetMapping
+    public ResponseEntity<UsuarioDTO> listar() {
+        return usuarioServices.getAllUsuarios();
+    }
+
+}

--- a/backend/abertoagora/src/main/java/br/com/abertoagora/model/EstabelecimentoElastic.java
+++ b/backend/abertoagora/src/main/java/br/com/abertoagora/model/EstabelecimentoElastic.java
@@ -17,7 +17,7 @@ import org.springframework.data.elasticsearch.annotations.Document;
 public class EstabelecimentoElastic {
 
     @Id
-    private String id;
+    private String idEstabelecimento ;
 
     private String cnpj;
     private String nomeEstabelecimento;

--- a/backend/abertoagora/src/main/java/br/com/abertoagora/repository/jpa/UsuarioRepository.java
+++ b/backend/abertoagora/src/main/java/br/com/abertoagora/repository/jpa/UsuarioRepository.java
@@ -1,4 +1,4 @@
-package br.com.abertoagora.repository;
+package br.com.abertoagora.repository.jpa;
 
 import br.com.abertoagora.model.Usuario;
 import org.springframework.data.jpa.repository.JpaRepository;
@@ -22,5 +22,7 @@ public interface UsuarioRepository extends JpaRepository<Usuario, Long> {
     Optional<Usuario> findByEmailUsuario (String emailUsuario);
 
     List<Usuario> findByDataCadastro (LocalDate dataCadastro);
+
+    List<Usuario> findAll();
 
 }

--- a/backend/abertoagora/src/main/java/br/com/abertoagora/services/EstabelecimentoServices.java
+++ b/backend/abertoagora/src/main/java/br/com/abertoagora/services/EstabelecimentoServices.java
@@ -1,6 +1,6 @@
 package br.com.abertoagora.services;
 
-import br.com.abertoagora.repository.EstabelecimentoElasticRepository;
+import br.com.abertoagora.repository.elastic.EstabelecimentoElasticRepository;
 import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;

--- a/backend/abertoagora/src/main/java/br/com/abertoagora/services/UsuarioServices.java
+++ b/backend/abertoagora/src/main/java/br/com/abertoagora/services/UsuarioServices.java
@@ -1,19 +1,62 @@
 package br.com.abertoagora.services;
 
-import br.com.abertoagora.repository.UsuarioRepository;
+import br.com.abertoagora.dto.UsuarioDTO;
+import br.com.abertoagora.model.Usuario;
+import br.com.abertoagora.repository.jpa.UsuarioRepository;
 import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
 
 @Service
 public class UsuarioServices {
     private final UsuarioRepository usuarioRepository;
     private final ModelMapper modelMapper;
 
-
     @Autowired
     public UsuarioServices(UsuarioRepository usuarioRepository, ModelMapper modelMapper) {
         this.usuarioRepository = usuarioRepository;
         this.modelMapper = modelMapper;
+    }
+
+    public ResponseEntity<UsuarioDTO> getByIdUsuario(Long idUsuario) {
+        Optional<Usuario> usuario = usuarioRepository.findByIdUsuario(idUsuario);
+        return usuario.map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    public ResponseEntity<UsuarioDTO> getByCpfUsuario(String cpfUsuario) {
+        Optional<Usuario> usuario = usuarioRepository.findByCpfUsuario(cpfUsuario);
+        return usuario.map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    public ResponseEntity<UsuarioDTO> getByNomeUsuario (String nomeUsuario) {
+        List<Usuario> usuarios = usuarioRepository.findByNomeUsuario(nomeUsuario);
+        return usuarios.stream().map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .findAny().orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    public ResponseEntity<UsuarioDTO> getByEmailUsuario (String emailUsuario) {
+        Optional<Usuario> usuario = usuarioRepository.findByEmailUsuario(emailUsuario);
+        return usuario.map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    public ResponseEntity<UsuarioDTO> getByDataCadastro (LocalDate dataCadastro) {
+        List<Usuario> usuarios = usuarioRepository.findByDataCadastro(dataCadastro);
+        return usuarios.stream().map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .findAny().orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    public ResponseEntity<UsuarioDTO> getAllUsuarios() {
+        List<Usuario> usuarios = usuarioRepository.findAll();
+        return usuarios.stream().map(value -> ResponseEntity.ok().body(modelMapper.map(value, UsuarioDTO.class)))
+                .findAny().orElseGet(() -> ResponseEntity.notFound().build());
     }
 }

--- a/backend/abertoagora/src/main/resources/application.properties
+++ b/backend/abertoagora/src/main/resources/application.properties
@@ -5,7 +5,7 @@ spring.datasource.username=root
 spring.datasource.password=root
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
-spring.elasticsearch.uris=http://elasticSearch:9200
-spring.elasticsearch.username=elastic
-spring.elasticsearch.password=AnSTXrcuYAL5kSc8Wsjq
+spring.data.elasticsearch.client.reactive.endpoints=elasticsearch:9200
+spring.elasitcsearch.name=elasticsearch
 elasticsearch.enabled=true
+spring.docker.compose.readiness-timeout=PT5M

